src/vendor_upstream/dom/ReactWheelHandler.js

```diff
@@ -63,12 +64,14 @@ class ReactWheelHandler {
     if (this._preventDefault) {
       event.preventDefault();
     }
-
+    
     var normalizedEvent = normalizeWheel(event);
-
+    
     // if shift is held, swap the axis of scrolling.
     if (event.shiftKey && ReactWheelHandler._allowInternalAxesSwap()) {
       normalizedEvent = ReactWheelHandler._swapNormalizedWheelAxis(normalizedEvent);
+    } else {
+      normalizedEvent.pixelX *= Locale.DIR_SIGN();
```

Comment:
shouldn't we have this outside the else statement?

Reply:
This one is a bit of an odd scenario. We'd only want to flip the sign in the case that shift is not pressed. Horizontal mouse scrolling should behave the same regardless of text direction.

In the case that shift _is_ pressed, we should maintain the scroll direction from `normalizedEvent.pixelY`, as it moves in the relative correct direction (down = positive, up = negative).

Let me know if I'm not making sense.